api/storage/client.go

```diff
@@ -147,3 +147,16 @@ func (c *Client) AddToUnit(storages []params.StorageAddParams) ([]params.ErrorRe
 	}
 	return out.Results, nil
 }
+
+// Destroy destroys specified storage entities.
+func (c *Client) Destroy(tags []names.Tag) ([]params.ErrorResult, error) {
```

Comment:
The value of this has never been explained. Again, I cannot see any benefit.

The issue I have with moving the logic into the API is that this is really a user-input / presentation issue. The "filesystem-" and "volume-" prefixes are there only to disambiguate overlapping ID domains. They are not part of any kind of canonical definition of the IDs, so don't really belong in the API layer. The CLI is all about user-input / presentation, hence why I put it there.

I'd really like to have the IDs be unambiguous now, but I'm not sure how we can fix that.